doc/catalyst-cli/catalyst-cli.rst

```diff
@@ -0,0 +1,241 @@
+
+For example, if we wanted to specify two pass pipelines, ``pipe1`` and ``pipe2``, where ``pipe1``
+applies the passes ``split-multiple-tapes`` and ``apply-transform-sequence``, and where ``pipe2``
+applies the pass ``inline-nested-module``, we would specify this pipeline configuration as:
+
+.. code-block::
+
+    --catalyst-pipeline=pipe1(split-multiple-tapes;apply-transform-sequence),pipe2(inline-nested-module)
+
+``--workspace=<path>``
+""""""""""""""""""""""
+
+The workspace directory where intermediate files are saved. The default is the current working
+directory.
+
+Note that the workspace directory must exist before running ``catalyst-cli`` with this option.
+
+``--module-name=<name>``
+""""""""""""""""""""""""
+
+The module name used in naming the output file(s). The default is ``"catalyst_module"``. Using the
+``-o`` option to specify the output filename overrides this option.
+
+``--async-qnodes[=<true|false>]``
+"""""""""""""""""""""""""""""""""
+
+Enable asynchronous QNodes.
+
+``--checkpoint-stage=<stage name>``
+"""""""""""""""""""""""""""""""""""
+
+Define a *checkpoint stage*, used to indicate that the compiler should start only after reaching the
+given pass.
+
+``--dump-catalyst-pipeline[=<true|false>]``
+"""""""""""""""""""""""""""""""""""""""""""
+
+Print (to stderr) the pipeline(s) that will be run.
+
+Examples
+^^^^^^^^
+
+To illustrate how to use the Catalyst CLI tool, consider the simple MLIR code, ``my_circuit.mlir``,
+which defines a function ``my_circuit`` that implements a single-qubit quantum circuit that applies
+the sequence of gates :math:`R_x(\theta) \to H \to H \to R_x(\theta)` to the input qubit for some
+rotation angle :math:`\theta`:
+
+.. code-block:: mlir
+
+    module {
+      func.func @my_circuit(%in_qubit: !quantum.bit, %angle: f64) -> !quantum.bit {
+        %0 = quantum.custom "RX"(%angle) %in_qubit : !quantum.bit
+        %1 = quantum.custom "Hadamard"() %0 : !quantum.bit
+        %2 = quantum.custom "Hadamard"() %1 : !quantum.bit
+        %3 = quantum.custom "RX"(%angle) %2 : !quantum.bit
+        return %3 : !quantum.bit
+      }
+    }
+
+We'll use the Catalyst CLI tool to run the ``quantum-opt`` compiler to perform the MLIR-level
+optimizations and translate the input to the LLVM dialect. We'll define a pass pipeline that applies
+two quantum-optimization passes:
+
+#. ``remove-chained-self-inverse``, which removes any operations that are applied next to their
+   (self-)inverses or adjoint, in this case the two adjacent Hadamard gates.
+#. ``merge-rotations``, which combines rotation gates of the same type that act sequentially, in
+   this case the two RX gates the become adjacent after the two Hadamard gates have been removed by
+   the ``remove-chained-self-inverse`` pass.
+
+To define the pass pipeline, we must supply the name of the function to which each pass applies
+using the ``func-name`` argument. To apply these two passes to our ``my_circuit`` function, we can
```

Comment:
Maybe it should be clarified here that the `func-name` argument is an argument specific to the two passes in the example, not universal to all passes or the catalyst-cli tool as a whole.

Reply:
(Final nitpick, I promise. All else looks good :)